### PR TITLE
KAFKA-10887 Migrate log4j-appender module to JUnit 5

### PR DESCRIPTION
**Pull request by g1geordie**
Migrate log4j-appender module to JUnit 5

### CODE REVIEW

**Comment by g1geordie** [comment]
@chia7712  please help me take a look

File: build.gradle [hunk 1]
```diff
@@ -1782,8 +1782,8 @@ project(':log4j-appender') {
     compile libs.slf4jlog4j
 
     testCompile project(':clients').sourceSets.test.output
-    testCompile libs.junitJupiterApi
-    testCompile libs.junitVintageEngine
+    testCompile libs.junitJupiter
+    testCompile libs.hamcrest
```
Comment (chia7712) [inline build.gradle:1786]: @ijuma Should we remove hamcrest? Removing it causes a bigger patch. The benefit of removing it is that our test assert get more consistent.

Comment (chia7712) [reply build.gradle:1786]: see https://github.com/apache/kafka/pull/9520#issuecomment-757426244 so keeping ```libs.hamcrest``` is fine.

**Review by chia7712 — APPROVED** [review]
@g1geordie LGTM. Thanks for your patch.

File: log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java [hunk 2]
```diff
@@ -130,55 +116,51 @@ public void testLog4jAppends() {
...
+    public void testSyncSendWithoutIgnoringExceptionsShouldNotThrowException() {
         Properties props = getLog4jConfig(true);
         props.put("log4j.appender.KAFKA.IgnoreExceptions", "false");
         PropertyConfigurator.configure(props);
 
         MockKafkaLog4jAppender mockKafkaLog4jAppender = getMockKafkaLog4jAppender();
         replaceProducerWithMocked(mockKafkaLog4jAppender, true);
 
-        logger.error(getMessage(0));
+        assertDoesNotThrow(() -> logger.error(getMessage(0)));
```
Comment (ijuma) [inline log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java:143]: What is the reason for having `assertDoesNotThrow` here and below? The test will fail if an exception is thrown, so seems like unnecessary noise.

Comment (chia7712) [reply log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java:143]: I considered it is a kind of highlight. However, it is unnecessary in this test case as it is the final line. @g1geordie WDYT? Should we file a PR to remove this unnecessary noise?

File: build.gradle (hunk 1 above)
Comment (g1geordie) [reply build.gradle:1786]: Thank you .
You are so strong that you can track a lot of issues simultaneously !

File: log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java (hunk 2 above)
Comment (g1geordie) [reply log4j-appender/src/test/java/org/apache/kafka/log4jappender/KafkaLog4jAppenderTest.java:143]: As you think .  I just point out where we do the check .
I will file a MINOR to change that    . Thank you

**Pull request merged**